common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -979,81 +1092,115 @@ private boolean ensureThreadStarted(int oldState) {
     }
 
     private void doStartThread() {
-        assert thread == null;
         executor.execute(new Runnable() {
             @Override
             public void run() {
+                processingLock.lock();
```

Comment:
is it a lock or try lock here, failing if it's false?

Reply:
A lock... we want to ensure we only process when the other thread "dies".